website/src/theme/TOC/index.tsx

```
@@ -0,0 +1,62 @@
+import TOCItems from '@theme/TOCItems';
+import type { Props } from '@theme/TOC';
+import styles from './styles.module.css';
+
+const LINK_CLASS_NAME = 'table-of-contents__link toc-highlight';
+const LINK_ACTIVE_CLASS_NAME = 'table-of-contents__link--active';
+
+// Safe hook to get blog post metadata.
+function useBlogPostSafe() {
+  try {
+    const { useBlogPost } = require('@docusaurus/plugin-content-blog/client');
+    return useBlogPost();
```

Biome (2.1.2)

[error] This hook is being called conditionally, but all hooks must be called in the exact same order in every component render.

For React to preserve state between calls, hooks needs to be called unconditionally and always in the same order.
See https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

(lint/correctness/useHookAtTopLevel)